network/wsPeer.go

```diff
@@ -229,6 +230,12 @@ type wsPeer struct {
 	// is present in wn.peers.
 	peerIndex int
 
+	// the peer's identity key which it uses for identityChallenge exchanges
+	identity         crypto.PublicKey
+	identityVerified uint32
+	// the identityChallenge is recorded to the peer so it may verify its identity at a later time
+	identityChallenge [32]byte
```

Comment:
the identityVerified value comes directly from the identity scheme's `VerifyResponse` function, which is called by tryConnect. tryConnect is already holding onto the `identityChallengeBytes` that will eventually be loaded into the peer.

Lots of patterns we could employ to wrap that up a bit and keep it as a single field, but managing these two separately is pretty simple (especially since in cases where tryConnect confirms the identity is verified, it does not need to store any challenge bytes on the peer, the challenge is over)